main.rb

```diff
@@ -66,3 +66,73 @@ def correct_name
 puts capitalized_person.correct_name
 capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
 puts capitalized_trimmed_person.correct_name
+
+class Classroom
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
+  end
+end
+
+class Student
+  attr_accessor :name, :classroom
+  attr_reader :rentals
+
+  def initialize(name)
+    @name = name
+    @classroom = nil
+    @rentals = [] # Initialize the rentals array
+  end
+end
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = [] # Initialize the rentals array
+  end
+end
+
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
```

Comment:
- [ ]  The Person class is missing. Kindly define a Person class whish should be similar to:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
    @rentals = []
  end

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```